### PR TITLE
Platform/13269/log4j formatting error

### PR DESCRIPTION
**Pull request by david-navapbc**
This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. *Include steps to test these changes*

execute `./prime fhirpath --help` and see no formatting errors from the logger

## Changes
- *Include a comprehensive list of changes in this PR*
added a call to the transformer plugin baked into the shade library we're using. This plugin is designed to elegantly remediate the issue described in the bug ticket. I additionally ensured the stripped-down log4j1 code that was making it's way into the phat-jar to not be there anymore. Lastly I made an update to the json template file in response to a deprecation warning. 

- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/build.gradle.kts [hunk 1]
```diff
@@ -375,6 +375,13 @@ tasks.jar {
 tasks.shadowJar {
     // our fat jar is getting fat! Or over 65K files in this case
     isZip64 = true
+    // this package is tied to log4j1. it's coming in not as a dependency but somehow as source from one of the
+    // dependencies in the project. it's (for now) harmless but flotsam so removing it. The call to transform(...)
+    // addresses the bug.
+    // for all the gory details see:
+    // https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/13269
+    exclude("org/apache/log4j/**")
+    transform(com.github.jengelman.gradle.plugins.shadow.transformers.Log4j2PluginsCacheFileTransformer())
```
Comment (jalbinson) [inline prime-router/build.gradle.kts:384]: Very interesting! A much deeper dive than I expected.

**Review by jalbinson — APPROVED** [review]

**Comment by jalbinson** [comment]
You should fill out the various details of the PR template just for the documentation. Most of it won't be relevant and you can link to the ticket for your deep dive.

**Review by arnejduranovic — APPROVED** [review]

File: prime-router/build.gradle.kts [hunk 2]
```diff
@@ -375,6 +375,13 @@ tasks.jar {
 tasks.shadowJar {
     // our fat jar is getting fat! Or over 65K files in this case
     isZip64 = true
+    // this package is tied to log4j1. it's coming in not as a dependency but somehow as source from one of the
+    // dependencies in the project. it's (for now) harmless but flotsam so removing it. The call to transform(...)
+    // addresses the bug.
```
Comment (arnejduranovic) [inline prime-router/build.gradle.kts:380]: What bug? -- so this comment makes sense in the future

Comment (david-navapbc) [reply prime-router/build.gradle.kts:380]: I'm happy to change the comment if you'd like. I would also like to point out the link to the bug ticket is directly below. Is the link sufficient or do you want another sentence describing the bug as well?

**Pull request merged**